libraries/Microsoft.Bot.Connector/Authentication/BotFrameworkAuthentication.cs

```diff
@@ -14,6 +14,28 @@ namespace Microsoft.Bot.Connector.Authentication
     /// </summary>
     public abstract class BotFrameworkAuthentication
     {
+        /// <summary>
+        /// Check whether the channel is Government.
+        /// </summary>
+        /// <returns>true if Government; false otherwise.</returns>
+        public abstract bool IsGovernment();
+
+        /// <summary>
+        /// Gets the App password for given App id.
+        /// </summary>
+        /// <param name="appId">The App id.</param>
+        /// <param name="cancellationToken">A cancellation token.</param>
+        /// <returns>The App password.</returns>
+        public abstract Task<string> GetAppPasswordAsync(string appId, CancellationToken cancellationToken);
```

Comment:
Certificates are supported in the old auth code, so please remove this code from the abstract class, as BFAuthentication subclasses designed for certificate support will not use this method.